modules/rubiconBidAdapter.js

```diff
@@ -22,6 +24,10 @@ const VIDEO_ENDPOINT = '//fastlane-adv.rubiconproject.com/v1/auction/video';
 
 const TIMEOUT_BUFFER = 500;
 
+var ACCOUNT_ID;
+getConfig('rubiconAdapter', config => ACCOUNT_ID = config.accountId);
```

Comment:
Yeah, that makes sense. I also agree that there's an architectural problem with prebid right now, and your diagram distills it down well.

One obvious fix that jumps out at me is if we moved towards an API like `requestBids(config)` instead, and then made `requestBids` pass the config through to any adapters. That would also move us one step closer to multiple auctions, since each auction would have its own state.

I know fixing all that is outside the scope of this PR... but I wonder if we could get things a little closer to that ideal, without too much effort?

For example, right now your modules look like:

```
getConfig();
adapter = makeAdapter(config);
adaptermanager.registerBidAdapter(adapter, ...);
```

Instead, we could do something like:

```
function adapterLoader(config) {
  return makeAdapter(config);
}
adaptermanager.registerBidAdapterLoader(adapterLoader, ...);
```

Then load the config inside `registerBidAdapterLoader`, and forward the call through to `registerBidAdapter` in the config callback.

Main benefits being:
- The circular dependence stays within prebid-core, and out of the modules
- Adapters would be easier to unit test (can send through mock configs)


Another benefit (disjoint from the first... so take it as an independent suggestion) is that you could namespace the global config. For example, if the bidder code is `rubicon`, then `registerAdapterLoader` function could decide (universally) that `rubiconAdapter` is where your config data goes. Then when it actually executes your `load` function, it can send that data right back to you.

Benefits there being:
- Lower publisher learning curve (config naming conventions consistent across all adapters)
- Adapters can't trample each others' config values accidentally